conformance/lang/expressions/relational-expr/greater_than_expression.balt

```diff
@@ -0,0 +1,1283 @@
+}
+
+Test-Case: output
+Description: Test greater than expression for user defined sub types of positive optional int
+Labels: DecimalNumber, module-type-defn, optional-type, relational-expr, singleton-type, union-type
+
+type Ints -2|-1|0|1|2;
+
+function init() {
+   Ints? a = 1;
+   Ints? b = 2;
+   Ints? c = 0;
+
+   io:println(a > b); // @output false
+   io:println(b > a); // @output true
+   io:println(a > c); // @output true
+   io:println(c > a); // @output false
+   io:println(b > c); // @output true
+   io:println(c > b); // @output false
+}
+
+Test-Case: output
+Description: Test greater than expression for user defined sub types of negative optional int
+Labels: DecimalNumber, module-type-defn, optional-type, relational-expr, singleton-type,
+        unary-minus, union-type
+
+type Ints -2|-1|-3|1|2;
+
+function init() {
+   Ints? a = -1;
+   Ints? b = -2;
+   Ints? c = -3;
+
+   io:println(a > b); // @output true
+   io:println(b > a); // @output false
+   io:println(a > c); // @output true
+   io:println(c > a); // @output false
+   io:println(b > c); // @output true
+   io:println(c > b); // @output false
+}
+
+Test-Case: output
+Description: Test greater than expression for user defined sub types where one operand is optional positive int and other one is optional negative int
+Labels: DecimalNumber, module-type-defn,optional-type, relational-expr, singleton-type,
+        unary-minus, union-type
+
+type Ints -2|-1|0|1|2;
+
+function init() {
+   Ints? a = 1;
+   Ints? b = 2;
+   Ints? c = -1;
+   Ints? d = -2;
+
+   io:println(a > c); // @output true
+   io:println(c > a); // @output false
+   io:println(b > c); // @output true
+   io:println(c > b); // @output false
+   io:println(a > d); // @output true
+   io:println(d > a); // @output false
+   io:println(d > b); // @output false
+   io:println(b > d); // @output true
+}
+
+Test-Case: output
+Description: Test greater than expression for different optional user defined sub types of int
+Labels: DecimalNumber, int:Signed8, int:Unsigned32, module-type-defn, optional-type, relational-expr, union-type
+
+const ONE = 1;
+const TWO = 2;
+type OneOrTwo ONE|TWO;
+type Ints int:Signed8|int:Unsigned32;
+
+function init() {
+   int:Signed8|int:Unsigned32? b = 13;
+   Ints? c = -23;
+   OneOrTwo? d = 1;
+
+   io:println(b > c); // @output true
+   io:println(b > d); // @output true
+   io:println(c > b); // @output false
+   io:println(c > d); // @output false
+   io:println(d > b); // @output false
+   io:println(d > c); // @output true
+}
+
+Test-Case: output
+Description: Test greater than expression when either or both operands are nil-literal
+Labels: nil-literal, optional-type, relational-expr
+
+function init() {
+   int? a = ();
+   int? b = 5;
+
+   io:println(a > b); // @output false
+   io:println(b > a); // @output false
+   io:println(() > ()); // @output false
+}
+
+Test-Case: output
+Description: Test greater than expression when either or both operands are boolean
```

Comment:
We don't need to say `either` here right? since both operands are boolean. 
```suggestion
Description: Test greater than expression when both operands are boolean
```
May be we can test when one is `boolean` and the other one is `boolean?`

Need to check the same in other operations as well.

Reply:
A compiler error is raised when operand is optional and other one is not, Have created an issue and added fail issue tests

Reply:
but we don't have fail issue tests when one operand is `boolean` and the other is `boolean?` right?